adoption/src/main/java/org/dcsa/conformance/standards/adoption/action/SupplyScenarioParametersAction.java

```diff
@@ -0,0 +1,104 @@
+
+@Getter
+public class SupplyScenarioParametersAction extends ConformanceAction {
+  public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+  private SuppliedScenarioParameters suppliedScenarioParameters = null;
+  private final LinkedHashSet<FilterParameter> filterParameters;
+
+  public SupplyScenarioParametersAction(
+      String publisherPartyName, FilterParameter... filterParameters) {
+    super(
+        publisherPartyName,
+        null,
+        null,
+        "SupplyScenarioParameters(%s)"
+            .formatted(
+                Arrays.stream(filterParameters)
+                    .map(FilterParameter::getQueryParamName)
+                    .collect(Collectors.joining(", "))));
+    this.filterParameters =
+        Stream.of(filterParameters).collect(Collectors.toCollection(LinkedHashSet::new));
+  }
+
+  @Override
+  public void reset() {
+    super.reset();
+    suppliedScenarioParameters = null;
+  }
+
+  @Override
+  public ObjectNode exportJsonState() {
+    ObjectNode jsonState = super.exportJsonState();
+    if (suppliedScenarioParameters != null) {
+      jsonState.set("suppliedScenarioParameters", suppliedScenarioParameters.toJson());
+    }
+    return jsonState;
+  }
+
+  @Override
+  public void importJsonState(JsonNode jsonState) {
+    super.importJsonState(jsonState);
+    if (jsonState.has("suppliedScenarioParameters")) {
+      suppliedScenarioParameters =
+          SuppliedScenarioParameters.fromJson(jsonState.required("suppliedScenarioParameters"));
+    }
+  }
+
+  @Override
+  public ObjectNode asJsonNode() {
+    ObjectNode objectNode = super.asJsonNode();
+    ArrayNode jsonFilterParameters = objectNode.putArray("FilterParametersQueryParamNames");
+    filterParameters.forEach(
+        filterParameter -> jsonFilterParameters.add(filterParameter.getQueryParamName()));
+    return objectNode;
+  }
+
+  @Override
+  public String getHumanReadablePrompt() {
+    return "Use the following format to provide the values of the specified query parameters"
+        + " for which your party can successfully process a GET request:";
+  }
+
+  @Override
+  public JsonNode getJsonForHumanReadablePrompt() {
+    return SuppliedScenarioParameters.fromMap(
+            filterParameters.stream()
+                .collect(
+                    Collectors.toMap(
+                        Function.identity(),
+                        filterParameter ->
+                            switch (filterParameter) {
+                              case INTERVAL -> "day";
+                              case DATE -> LocalDateTime.now().format(DATE_FORMAT);
+                            })))
+        .toJson();
+  }
+
+  @Override
+  public boolean isInputRequired() {
+    return true;
+  }
+
+  @Override
+  public void handlePartyInput(JsonNode partyInput) {
+    super.handlePartyInput(partyInput);
+    suppliedScenarioParameters = SuppliedScenarioParameters.fromJson(partyInput.get("input"));
```

Comment:
Possibly NPE. The `partyInput.get` can return  `null` and the callee does not seem to support it.

If you are sure it is never null, use `.required`. You can also use `.path` if "defaults" is fine for the callee.

As far as I can see, use of `JsonNode.get` instead of `.path` or `.required` is "almost always" a trap.

Reply:
@gj0dcsa, just to make sure I got you right: I just checked, and if I supply `"interval": "asdf"` that also still makes it continue and send it to the user. Is it the right place to add validation of this input data, in this method? If you have an example otherwise, please point it out.